### ISSUE TITLE
Why is the performance poor?

### ISSUE THREAD

**Issue by zqxie77** [Visuang/VLD#2]
CUDA_VISIBLE_DEVICES=1 python main.py --mode train --dataset bupt --pid_num 1074 --output_path logs/bupt --train_bs 2
===> MCPRL-ReID Dataset (train) <===
Number of identities: 1074
Number of samples   : 7148
Resized position embedding: %s to %s torch.Size([197, 768]) torch.Size([163, 768])
Position embedding resize to height:18 width: 9
Existed dirs: logs/bupt
Existed dirs: logs/bupt/models/
Existed dirs: logs/bupt/logs/

Namespace(cuda='cuda', mode='train', dataset='bupt', vcm_data_path='data/HITSZ-VCM/', bupt_data_path='data/BUPTCampus/', batch_size=4, num_pos=4, is_STA=True, img_w=144, img_h=288, test_batch=32, seed=1, pid_num=1074, steps=200, learning_rate=2.5e-05, lr_times=25, num_workers=8, weight_decay=0.0001, output_path='logs/bupt', total_train_epoch=24, eval_epoch=2, resume_test_path='/data1/ls/code/VIV_ReID/Github_vld/VLD/bupt/debug/VLD_vcm/models', STH_start_layer=9, train_frame_sample='random', sequence_length=6, random_flip=True, fake=False, test_frame_sample='uniform', test_sampler='ConsistentModalitySampler', test_bs=64, train_sampler='RandomIdentitySampler', train_bs=2, train_sampler_nc=2, train_sampler_nt=1, max_rank=20, weight_a=0.08, weight_b=0.4, weight_c=1.0)
Time: 2025-06-06 01:56:26; Epoch: 0; pid_loss: 6.9795065;  pid_loss_proj: 6.9793844;  tri_loss: 9.612525;  tri_loss_proj: 2.868162;  tri_loss_last: 1.9511213;  ide_loss_text: 0.6840136;  pid_loss_STP: 2.791818;  tri_loss_STP: 3.0550587;  ; 2.5000000000000004e-07
Time: 2025-06-06 01:57:53; Epoch: 1; pid_loss: 6.9798193;  pid_loss_proj: 6.9793067;  tri_loss: 1.0479273;  tri_loss_proj: 0.7671565;  tri_loss_last: 0.8116042;  ide_loss_text: 0.7628833;  pid_loss_STP: 2.791771;  tri_loss_STP: 0.8136373;  ; 8.5e-06
Extracting Query Feature...
Extracting Gallery Feature...
Time: 2025-06-06 02:00:50; Test on Dataset: bupt, 
 task: RGB->IR :  mAP: 0.32% | Rank-1: 0.00% | Rank-5: 0.39% | Rank-10: 0.98% | Rank-20: 2.15%., 
 task: IR ->RGB:  mAP: 0.70% | Rank-1: 0.19% | Rank-5: 0.57% | Rank-10: 1.34% | Rank-20: 2.49%.
Time: 2025-06-06 02:02:16; Epoch: 2; pid_loss: 6.9808836;  pid_loss_proj: 6.9788475;  tri_loss: 0.73996764;  tri_loss_proj: 0.70509315;  tri_loss_last: 0.7270257;  ide_loss_text: 0.61771864;  pid_loss_STP: 2.7915533;  tri_loss_STP: 0.7222242;  ; 1.675e-05
Time: 2025-06-06 02:03:42; Epoch: 3; pid_loss: 6.9933004;  pid_loss_proj: 6.994712;  tri_loss: 0.71322256;  tri_loss_proj: 0.698381;  tri_loss_last: 0.7105788;  ide_loss_text: 0.60143256;  pid_loss_STP: 2.7997675;  tri_loss_STP: 0.70786476;  ; 2.4058009214827173e-05
Extracting Query Feature...
Extracting Gallery Feature...
Time: 2025-06-06 02:04:40; Test on Dataset: bupt, 
 task: RGB->IR :  mAP: 0.31% | Rank-1: 0.00% | Rank-5: 0.59% | Rank-10: 0.98% | Rank-20: 1.76%., 
 task: IR ->RGB:  mAP: 0.30% | Rank-1: 0.19% | Rank-5: 0.38% | Rank-10: 0.77% | Rank-20: 1.34%.
Time: 2025-06-06 02:06:07; Epoch: 4; pid_loss: 6.988642;  pid_loss_proj: 6.9894967;  tri_loss: 0.7048449;  tri_loss_proj: 0.6962002;  tri_loss_last: 0.7051187;  ide_loss_text: 0.5996067;  pid_loss_STP: 2.7964745;  tri_loss_STP: 0.7029277;  ; 2.334206437183243e-05

**Comment by Visuang** [comment]
Please carefully check the random sampling settings for the BUPT dataset. When train_bs=2, the sampling becomes infeasible. Increasing train_bs leads to better performance, and the current results are obtained using the default train_bs setting.

**Comment by zqxie77** [comment]
Thanks a lot for the helpful feedback and great work! 
I’d like to confirm the specific experimental environment settings.Would you mind sharing the details when you have a moment?
